### PR TITLE
Pull in failed 5-class models, update 3-class with feature importance

### PR DESCRIPTION
**Pull request by steeleb**
Hey Caitlin, this looks huge, but it's actually pretty repetitive and short, lots of commented out code sections. 

There are two things happening here:
1) I've added in the original 5-class models and performance metrics (these performed super poorly), just for good measure. You'll notice that about 1/3 of the way down the script I just comment out the code, because I'm not going to export models for a poorly performing model. 
2) I added in feature importance plots to all the scripts. In doing this, I altered the LS8/9/Sen2 models for the 3-class algos since the most important feature was aerosol, and that doesn't make sense at all for classifying *sediment* (though it makes sense for clouds, obviously). 

Script modeling/08_Train_Test_Split.Rmd should be nearly identical to modeling/08_Train_Test_Split_3class.Rmd. Turns out that issue you saw with Band 6/8 didn't matter, because I never referenced that object again. Doh. You'll see I removed a few rows of code with objects that never get used. 

For this review, please take a peek at scripts 09-13 for the 5-class models to check for silly mistakes, they are incredibly redundant, sorry. 

I'll add a few notes about other small changes along the way. 

Thanks again for your continued help with this.

### CODE REVIEW

File: modeling/13_Sentinel2_GTB_3class.Rmd [hunk 1]
```diff
@@ -210,8 +248,8 @@ sen = (ee.ImageCollection('COPERNICUS/S2_SR_HARMONIZED')
   .filter(ee.Filter.eq('GENERAL_QUALITY', 'PASSED'))
   .filter(ee.Filter.eq('GEOMETRIC_QUALITY', 'PASSED'))
   .filter(ee.Filter.eq('SENSOR_QUALITY', 'PASSED'))
-  .filter(ee.Filter.eq('DEGRADED_MSI_DATA_PERCENTAGE', 0))
-  .filter(ee.Filter.eq('SNOW_ICE_PERCENTAGE', 0))
+  .filter(ee.Filter.lt('DEGRADED_MSI_DATA_PERCENTAGE', 10))
+  .filter(ee.Filter.lt('SNOW_ICE_PERCENTAGE', 10))
   # masks
```
Comment (steeleb) [inline modeling/13_Sentinel2_GTB_3class.Rmd:253]: some intentional changes here, turns out these filters were way too conservative and wiped out about 70% of the tiles. Oops.

File: modeling/08_Train_Test_Split_3class.Rmd [hunk 2]
```diff
@@ -108,25 +108,15 @@ sen2_labels = read_rds("data/labels/S2_labels_for_tvt_2024-04-25.RDS") %>%
...
 ```
 
 ### Check for complete obs
 
 ```{r}
-ls57_bands_text = c("SR_B1", "SR_B2", "SR_B3", "SR_B4", "SR_B5", "SR_B7")
-ls89_bands_text = c("SR_B2", "SR_B3", "SR_B4", "SR_B5", "SR_B6", "SR_B7")
-sen_bands_text = c("SR_B1", "SR_B2", "SR_B3", "SR_B4", "SR_B5", "SR_B6", 
-                  "SR_B7", "SR_B8", "SR_B8A", "SR_B9", "SR_B11", "SR_B12")
-
```
Comment (steeleb) [inline modeling/08_Train_Test_Split_3class.Rmd:129]: ...these are those objects that are never referred to...

File: modeling/08_Train_Test_Split_3class.Rmd [hunk 3]
```diff
@@ -400,10 +390,6 @@ ee_s2_validate = sen_to_eeFeat(r.sen_validate)
 Define the input features and output labels
 
 ```{python}
-ls57_input_feat = ["SR_B1", "SR_B2", "SR_B3", "SR_B4", "SR_B5", "SR_B7"]
-ls89_input_feat = ["SR_B2", "SR_B3", "SR_B4", "SR_B5", "SR_B6", "SR_B7"]
-sen_input_feat = (["SR_B1", "SR_B2", "SR_B3", "SR_B4", "SR_B5", "SR_B6", 
-  "SR_B7", "SR_B8", "SR_B8A","SR_B9", 'SR_B11', 'SR_B12'])
```
Comment (steeleb) [inline modeling/08_Train_Test_Split_3class.Rmd:406]: more stuff that is never referred to

File: modeling/11_Landsat_8_GTB.Rmd [hunk 4]
```diff
@@ -0,0 +1,403 @@
...
+values = list(sorted_importance.values())
+
+# Plot the bar graph
+plt.figure(figsize=(10, 6))
+plt.barh(keys, values, color='skyblue')
+
+# Adding titles and labels
+plt.xlabel('Feature Importance')
+plt.ylabel('Band')
+plt.title('Feature importance for 3-class GTB model for Landsat 8')
```
Comment (ccmothes) [inline modeling/11_Landsat_8_GTB.Rmd:143]: 5-class?

**Review by ccmothes — APPROVED** [review]
All looks good to me! Just had one comment.

File: modeling/11_Landsat_8_GTB.Rmd (hunk 4 above)
Comment (steeleb) [reply modeling/11_Landsat_8_GTB.Rmd:143]: yuppers. fixing now!

Comment (steeleb) [inline modeling/11_Landsat_8_GTB.Rmd:143]:
```suggestion
plt.title('Feature importance for 5-class GTB model for Landsat 8')
```

**Comment by steeleb** [comment]
Thank you, @ccmothes.

**Pull request merged**